sdk/docs/manually-written/sdk/component-howtos/application-development/daml-codegen-java.rst

```diff
@@ -1,37 +1,134 @@
-.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
-.. SPDX-License-Identifier: Apache-2.0
-
 .. _daml-codegen-java:
 
+Daml Codegen for Java
+=====================
+
+This tool is part of the Daml Assistant and is used to generate Java classes for Daml data types.
+
+Overview
+-------------
+You can use the Daml codegen to generate Java classes representing Daml contract templates.
+These classes incorporate all boilerplate code for constructing corresponding ledger ``com.daml.ledger.api.v1.CreateCommand``,
+``com.daml.ledger.api.v1.ExerciseCommand``, ``com.daml.ledger.api.v1.ExerciseByKeyCommand``, and ``com.daml.ledger.api.v1.CreateAndExerciseCommand``.
+
+Download
+-------------
+
+The Daml codegen is part of the Daml Assistant. :ref:`Download the Daml Assistant <assistant-manual-managing-releases>`.
+
+.. TODO(#347) Fix the link to link to Download the Daml Assistant of "Daml Assistant"
+
+Install
+-------------
+
+The Daml codegen is part of the Daml Assistant.
+
+.. TODO(#347) Add link "Download the Daml Assistant" to point to download section of "Daml Assistant"
+
+Configure
+-------------
+
+There are two modes to configure the Daml codegen:
+
+- Command line configuration, specifying **all** settings in the command line.
+
+- Project file configuration, specifying **all** settings in the ``daml.yaml``.
```

Comment:
Note this reads more like reference documentation. Passive, factual voice. 

You can leave it though here, as it doesn't hurt too much here.

However, when writing new text then please follow https://diataxis.fr/how-to-guides/#the-language-of-how-to-guides

Reply:
rewording